### PR TITLE
jujutsu: refactor, add emily to maintainers

### PR DESCRIPTION
**Pull request by emilazy**
## Description of changes

This is critical to my workflow, so let’s step up to help out.

Probably these commits didn’t have to be so thoroughly factored out, but when Jujutsu makes it this easy, why not?

cc @bbigras – did you want to co‐maintain this?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bbigras** [comment]
> cc @bbigras – did you want to co‐maintain this?

Yeah, you can add me as a maintainer too. :)

File: pkgs/by-name/ju/jujutsu/package.nix [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      zstd
+      libgit2
+      libssh2
+    ]
+    ++ lib.optional (!stdenv.hostPlatform.isDarwin) [ openssl ]
```
Comment (misuzu) [inline pkgs/by-name/ju/jujutsu/package.nix:51]:
```suggestion
    ++ lib.optional (!stdenv.hostPlatform.isDarwin) openssl
```

```suggestion
    ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [ openssl ]
```

Comment (emilazy) [reply pkgs/by-name/ju/jujutsu/package.nix]: Thanks, typo 🫠 Fixed!

We need to lint for this…

**Comment by samueltardieu** [comment]
Result of `nixpkgs-review pr 339702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>jujutsu</li>
    <li>lazyjj</li>
  </ul>
</details>

**Review by samueltardieu — APPROVED** [review]

**Comment by samueltardieu** [comment]
> Probably these commits didn’t have to be so thoroughly factored out, but when Jujutsu makes it this easy, why not?

Indeed!

**Review by thoughtpolice — APPROVED** [review]
I absolutely despise every aesthetic choice made by nixfmt and absolutely reject it everywhere on my own codebases. But I'll live with it here I guess.

**Comment by emilazy** [comment]
I have plenty of bones to pick with the style, but it’ll get swept up in the treewide before 24.11 anyway, so might as well bite the bullet now :)

(I think the “new files must be formatted” CI lint would complain if I hadn’t done it, anyway, given the `pkgs/by-name` move.)

**Pull request merged**